### PR TITLE
Unity/tracking

### PR DESCRIPTION
**Pull request by mitcmurph**

### CODE REVIEW

File: unity/XRTProject/Assets/Scripts/Networking/ApiRequest.cs [hunk 1]
```diff
@@ -61,7 +61,7 @@ public ApiRequest HandleResponse(Action<object> responsePayload)
 
     public ApiRequest HandleJsonResponse<T>(Action<T> responsePayload)
     {
-        this.responsePayload = (p) => JsonConvert.DeserializeObject<T>(p);
+        this.responsePayload = (p) => responsePayload.Invoke(JsonConvert.DeserializeObject<T>(p));
```
Comment (mitcmurph) [inline unity/XRTProject/Assets/Scripts/Networking/ApiRequest.cs:64]: Was this missing?

File: unity/XRTProject/Assets/Scripts/Training/TrainingManager.cs [hunk 2]
```diff
@@ -82,6 +84,8 @@ private void Awake()
             {
                 CurrentUser = user;
                 OnUserReady?.Invoke();
+                Logger = gameObject.AddComponent<Logger>();
+                Logger.Setup(apiService, TrainingModule, user);
```
Comment (mitcmurph) [inline unity/XRTProject/Assets/Scripts/Training/TrainingManager.cs:88]: Best way to attach it to TrainingManager? Or should create an independent component?

File: unity/XRTProject/Assets/Scripts/Networking/ApiRequest.cs (hunk 1 above)
Comment (Agid-Kaharuba) [reply unity/XRTProject/Assets/Scripts/Networking/ApiRequest.cs:64]: Yes, that might actually solve the networking bug....thanks!

File: unity/XRTProject/Assets/Scripts/Training/TrainingManager.cs (hunk 2 above)
Comment (Agid-Kaharuba) [reply unity/XRTProject/Assets/Scripts/Training/TrainingManager.cs:88]: Probably better for you to make the Logger to be its own singleton independent from the training module. Then the training manager would access logger to setup.

Comment (mitcmurph) [reply unity/XRTProject/Assets/Scripts/Training/TrainingManager.cs:88]: Feedback from Agid. Make Logger a singleton and have TrainingManager set it up.

File: unity/XRTProject/Assets/Scripts/Training/TrainingManager.cs [hunk 3]
```diff
@@ -70,6 +70,7 @@ private void Awake()
         }
 
         apiService = new ApiService(token);
+        gameObject.AddComponent<Track>().Setup(apiService);
```
Comment (mitcmurph) [inline unity/XRTProject/Assets/Scripts/Training/TrainingManager.cs:73]: @Agid-Kaharuba Couldn't figure out how to implement without Monobehaviour/using async

**Pull request merged**